<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to scSpotlight • scSpotlight</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to scSpotlight">
<!-- Google tag (gtag.js) --><script async src="https://www.googletagmanager.com/gtag/js?id=G-5NHQW8GD3W"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-5NHQW8GD3W');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scSpotlight</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.3.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/scSpotlight.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/obenno/scSpotlight/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to scSpotlight</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/obenno/scSpotlight/blob/v0.0.4/vignettes/articles/scSpotlight.Rmd" class="external-link"><code>vignettes/articles/scSpotlight.Rmd</code></a></small>
      <div class="d-none name"><code>scSpotlight.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="seurat-analysis-workflow">Seurat Analysis Workflow<a class="anchor" aria-label="anchor" href="#seurat-analysis-workflow"></a>
</h2>
<p><code>scSpotlight</code> uses the Seurat standard single cell
analysis workflow, we recommend new users going through their <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial" class="external-link">pbmc3k
tutorial</a> to better understand each processing step. The
demonstration dataset used is a 2,700 Peripheral Blood Mononuclear Cells
(PBMCs) sample made publicly available by 10X Genomics. The matrix could
be downloaded <a href="https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz" class="external-link">here</a>.</p>
<p>A typical seurat analysis workflow is like below:</p>
<p><img src="images/seurat_standard_workflow.png" width="986"></p>
</div>
<div class="section level2">
<h2 id="invoke-app">Invoke App<a class="anchor" aria-label="anchor" href="#invoke-app"></a>
</h2>
<p>To start using scSpotlight, please use the command below and access
the app via: <code>127.0.0.1:8081</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scSpotlight</span><span class="fu">::</span><span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span>options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>port <span class="op">=</span> <span class="fl">8081</span>, host <span class="op">=</span> <span class="st">"0.0.0.0"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The default mode of app is <code>processing</code>. User could also
change mode to <code>viewer</code>, which will only allow illustrating
dataset and querying gene expressions:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scSpotlight</span><span class="fu">::</span><span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span>options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>port <span class="op">=</span> <span class="fl">8081</span>, host <span class="op">=</span> <span class="st">"0.0.0.0"</span><span class="op">)</span>, runningMode <span class="op">=</span> <span class="st">"viewer"</span><span class="op">)</span></span></code></pre></div>
<p>If one needs to load a very large dataset, use <code>dataDir</code>
parameter to mount data directory and load <code>Rds</code> file
directly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scSpotlight</span><span class="fu">::</span><span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span>options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>port <span class="op">=</span> <span class="fl">8081</span>, host <span class="op">=</span> <span class="st">"0.0.0.0"</span><span class="op">)</span>, runningMode <span class="op">=</span> <span class="st">"viewer"</span>, dataDir <span class="op">=</span> <span class="st">"/path/to/data_directory"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="main-panel">Main Panel<a class="anchor" aria-label="anchor" href="#main-panel"></a>
</h2>
<p>Main panel of <code>scSpotlight</code> will be like:</p>
<p><img src="images/main_panel.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="input-files">Input Files<a class="anchor" aria-label="anchor" href="#input-files"></a>
</h2>
<p>Under the “processing” mode, <code>scSpotlight</code> supports
compressed file containing typical 10X cellranger output (<a href="https://math.nist.gov/MatrixMarket/formats.html" class="external-link">Matrix Market
Format</a>):</p>
<pre><code>$ tar -tvf PBMC_demo.tar.gz
drwxrwxr-x xzx/xzx           0 2024-01-11 21:11 PBMC_demo/
-rw-r--r-- xzx/xzx    67972057 2024-01-11 21:10 PBMC_demo/matrix.mtx.gz
-rw-r--r-- xzx/xzx      294722 2024-01-11 21:10 PBMC_demo/features.tsv.gz
-rw-r--r-- xzx/xzx       46443 2024-01-11 21:10 PBMC_demo/barcodes.tsv.gz</code></pre>
<p>and processed Seurat object <code>seurat_processed.Rds</code> saved
by [Seurat::SaveSeuratRds()].</p>
<p>If using compressed matrix file as input, scSpotlight will process
data after loading <strong>without cell filtration</strong> and the
dimension reduction will be shown.</p>
</div>
<div class="section level2">
<h2 id="filter-cells">Filter Cells<a class="anchor" aria-label="anchor" href="#filter-cells"></a>
</h2>
<p>scSpotlight will automatically illustrate nGene (nFeature_RNA), nUMI
(nCount_RNA), mitochondrial gene UMI percentage (percent.mt) and
ribosomal protein gene UMI percentage (percent.rp) for each cell in
VlnPlot panel. One could inspect the distribution and roughly evaluate
the library quality. User could discard cells with too few or too many
features detected and dying cells with high mitochondrial UMI percentage
by adjusting parameters in the “Cell Filtering” block. The filtration
step will also update highly variable genes (HVGs), the dimension
reduction and the cell clustering result.</p>
<div class="figure">
<img src="images/vlnplot_1.png" alt="VlnPlot summarizes nGene, nUMI, percent.mt and percent.rp distribution" width="1280"><p class="caption">
VlnPlot summarizes nGene, nUMI, percent.mt and percent.rp distribution
</p>
</div>
<div class="figure">
<img src="images/vlnplot_2.png" alt="VlnPlot full screen" width="1280"><p class="caption">
VlnPlot full screen
</p>
</div>
<div class="figure">
<img src="images/cellfilter.png" alt="Cell filtration block" width="1280"><p class="caption">
Cell filtration block
</p>
</div>
</div>
<div class="section level2">
<h2 id="highly-variable-features-and-clustering">Highly variable features and Clustering<a class="anchor" aria-label="anchor" href="#highly-variable-features-and-clustering"></a>
</h2>
<p>User will be able to adjust highly variable gene selection method and
parameters related to unsupervised clustering.</p>
<div class="figure">
<img src="images/updateCluster.png" alt="Clustering setting block" width="1280"><p class="caption">
Clustering setting block
</p>
</div>
</div>
<div class="section level2">
<h2 id="cell-identity-groups-visualization">Cell identity (groups) visualization<a class="anchor" aria-label="anchor" href="#cell-identity-groups-visualization"></a>
</h2>
<div class="figure">
<img src="images/group_by.png" alt="Select cell identity" width="1280"><p class="caption">
Select cell identity
</p>
</div>
</div>
<div class="section level2">
<h2 id="gene-expression">Gene expression<a class="anchor" aria-label="anchor" href="#gene-expression"></a>
</h2>
<p>Input gene symbol to view its expression aside of the original
cluster view</p>
<div class="figure">
<img src="images/single_gene_expression.png" alt="Single gene expression" width="1280"><p class="caption">
Single gene expression
</p>
</div>
<p>Input multiple genes to view all of them via Seurat::FeaturePlot(),
and click anyone of the subfigures to acitvate single gene view.</p>
<div class="figure">
<img src="images/multi_gene_expression.png" alt="Multiple gene expression" width="1280"><p class="caption">
Multiple gene expression
</p>
</div>
<p>Or view expression with Seurat::DotPlot():</p>
<div class="figure">
<img src="images/dotplot.png" alt="Gene expression DotPlot" width="1280"><p class="caption">
Gene expression DotPlot
</p>
</div>
</div>
<div class="section level2">
<h2 id="select-and-rename-cells">Select and rename cells<a class="anchor" aria-label="anchor" href="#select-and-rename-cells"></a>
</h2>
<video controls width="100%"><source src="https://github.com/obenno/scSpotlight/blob/main/vignettes/articles/images/select_rename_cells.webm?raw=true" type="video/webm" srcset="NA"></source></video>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhixia Xiao.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
